src/object.rs

```diff
@@ -178,23 +177,31 @@ impl JSObject {
             return Err(JSException {
                 value: JSValue {
                     raw: exception,
-                    ctx: context.raw,
+                    ctx: context,
                 },
             });
         }
 
         if result.is_null() {
             return Err(JSException {
-                value: JSValue::new_string(
-                    context,
-                    "Cannot call this object as a function: it is not a valid function",
-                ),
+                value: JSValue {
+                    raw: unsafe {
+                        sys::JSValueMakeString(
+                            context,
+                            JSString::from(
+                                "Cannot call this object as a function: it is not a valid function",
+                            )
+                            .raw,
+                        )
+                    },
+                    ctx: context,
+                },
```

Comment:
Why this change?

Reply:
Because it's not possible to use `JSValue::new_string` with a `JSContextRef`: only with a `JSContext`.

I'm about to write another patch to have a generic impl' for that.

Reply:
oh, right. Would help if I was awake (despite it being middle of the afternoon).